exercises/practice/palindrome-products/.meta/example.rs

```diff
@@ -1,69 +1,79 @@
-    }
-
-    /// Insert a new set of factors into an existing palindrome
-    pub fn insert(&mut self, mut a: u64, mut b: u64) {
-        if a > b {
-            std::mem::swap(&mut a, &mut b);
-        }
-        self.factors.push((a, b));
-        self.factors.sort_unstable();
-        self.factors.dedup();
+    /// Get the value of this palindrome.
+    pub fn into_inner(self) -> u64 {
+        return self.0;
     }
 }
 
-/// return the (min, max) palindrome pair comprised of the products of numbers in the input range
 pub fn palindrome_products(min: u64, max: u64) -> Option<(Palindrome, Palindrome)> {
-    let mut result = None;
-
-    for a in min..=max {
-        for b in min..=a {
-            if is_palindrome(a * b) {
-                result = match result {
-                    None => Some((Palindrome::new(a, b), Palindrome::new(a, b))),
-                    Some((mut minp, mut maxp)) => {
-                        match (a * b).cmp(&minp.value()) {
-                            std::cmp::Ordering::Greater => {}
-                            std::cmp::Ordering::Less => minp = Palindrome::new(a, b),
-                            std::cmp::Ordering::Equal => minp.insert(a, b),
+    let mut pmin: Option<Palindrome> = None;
+    let mut pmax: Option<Palindrome> = None;
+    for i in min..=max {
+        for j in i..=max {
+            let p = i * j;
+            if let Some(x) = Palindrome::new(p) {
+                match pmin {
+                    None => {
+                        pmin = Some(x);
+                    },
+                    Some(t) => {
+                        if t.0 > x.0 {
+                            pmin = Some(x);
                         }
-                        match (a * b).cmp(&maxp.value()) {
-                            std::cmp::Ordering::Less => {}
-                            std::cmp::Ordering::Greater => maxp = Palindrome::new(a, b),
-                            std::cmp::Ordering::Equal => maxp.insert(a, b),
+                    },
+                }
+                match pmax {
+                    None => {
+                        pmax = Some(x);
+                    },
+                    Some(t) => {
+                        if t.0 < x.0 {
+                            pmax = Some(x);
                         }
-                        Some((minp, maxp))
                     }
-                };
+                }
+            }
+        }
+    }
+    match pmax {
+        None => {
+            None
+        },
+        Some(r) => {
+            match pmin {
+                None => {
+                    panic!("something wrong");
+                },
+                Some(l) => {
+                    Some((l, r))
+                }
             }
         }
     }
-
-    result
 }
-
-#[inline]
-pub fn is_palindrome(n: u64) -> bool {
-    let s = n.to_string().into_bytes();
-    s.iter().zip(s.iter().rev()).all(|(a, b)| a == b)
+fn is_palindrome(v: u64) -> bool {
```

Comment:
Haha, this implementation totally bypasses the obvious string implementation. No need to change it.